doc/notes/cost-model-representations/cost-model-representations.md

```diff
@@ -0,0 +1,216 @@
+# Representations of the Plutus Core cost model
+
+The `plutus` codebase uses several different representations of the Plutus Core
+cost model.  It can be somewhat difficult to keep track of what all of these are
+used for; this document attempts to give an overview to clarify matters.  An
+essential point is that if you're using `mkEvaluationContext` then cost model
+parameters much be provided in the correct order, otherwise script costs
+will be inaccurate: see [`EvaluationContext`](#evaluationcontext) below.
+
+## Introduction
+
+The Plutus Core language incorporates a _cost model_ which assigns costs for CPU
+usage and memory usage to the evaluation of an Untyped Plutus Core program on
+the standard Plutus evaluator (the _CEK machine_).  CPU costs are measured in
+abstract units called `ExUnit`s (_execution units_), with one `ExUnit`
+corresponding to one picosecond on a specific dedicated benchmarking machine.
+We expect that that the actual time required to execute a program on some other
+machine will be (approximately) proportional to the total number of `ExUnit`s.
+
+Memory usage, or "size", is measured in units of 64-bit words (although with
+hindsight the number of bytes required to store an object might have been a
+better measure).
+
+The cost model consists of two components:
+
+  1. A CPU and memory cost for every basic step of the evaluator. 
+  2. A pair of _costing functions_ for each built-in function which give an
+     approximation to the CPU usage and memory usage of the function.  The cost
+     generally depends on the size of the input(s) to the builtin, which is why
+     we need a function rather than just a constant.  The CPU costing functions are
+     determined by running benchmarks and fitting regression models to the
+     results.  The memory costs of a built-in function (which measure only the
+     memory required to store the output of the function, and not any transient
+     memory allocated during execution) are determined by understanding what the
+     builtin does, and perhaps by examining the implementation.
+
+### Serialised representation
+
+Concrete values for the coefficients of the costing functions for the builtins
```

Comment:
Worth being clear that these are not the values on mainnet, since people get confused about that. Also this could go in the Haddock for the Haskell values that are the read-in versions of these files.